fastparquet/test/test_writer.py

```diff
@@ -233,6 +235,15 @@ def test_make_definitions_without_nulls():
     # halper, metadata = mock(), mock()
 
 
+def test_empty_row_group(tempdir):
+    fname = os.path.join(tempdir, 'temp.parq')
+    data = pd.DataFrame({'o': np.random.choice(['hello', 'world'],
+                                               size=1000)})
+    writer.write(fname, data, partitions=[0, 900, 1800])
+    pf = ParquetFile(fname)
+    assert len(pf.row_groups) == 2
```

Comment:
The first row-group has rows 0-899, the next has 900-1799, so actually 900-end, and the third row group does not get created.